### ISSUE TITLE
Add reading goal

### ISSUE THREAD

**Issue by knjk04** [knjk04/book-project#55]
**Describe the solution you'd like**
Add a page where the user can set a target for the number of the books they want to have read by the end of the year. 

The page should include
- The yearly target (e.g. 52 books to read by the of the year)
- A way to reduce or increase the number of books in that target
- How many books the user has read<sup>1</sup> during the year (so far if the year is not the past)
- Which books they have read during that year that count towards the target
- A way to see how many books the user needs to read a week on average in order to reach the target
- Whether they are ahead of target, behind or on target
- A visual element that shows what percentage of the way through they are towards that target

1. Books that have been marked as 'to read' that were started and finished in during the year count towards the year's reading goal. This should be made clear to the user.

**Additional context**

The reading goal should be an entity so that it is saved in the database (so that users can see previous yearly challenges they set for themselves).

In short, this reading target feature should be similar to Goodreads's.

**Comment by knjk04** [comment]
A text field has been added (as well as an entity, repository and a service on the back-end) to allow the user to set (and change) the number of books they want to read by the end of the year. Whether (and by how much) the user is ahead of behind schedule is shown. If the user has accomplished their target, a congratulatory message is displayed.

Still need to implement:
- Which books they have read during that year that count towards the target
- A visual element (e.g. a progress bar) that shows what percentage of the way through they are towards that target
- Store reading goals for different years. This could be down by storing the year that the goal is set. The UI needs to be adapted to show goals from previous years.

New issues will be created to address these.
